src/proxy/IReplicatedMap.ts

```diff
@@ -0,0 +1,171 @@
+import * as Promise from 'bluebird';
+import {DistributedObject} from '../DistributedObject';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+import Long = require('long');
+import {TimeUnit} from '../util/TimeUnit';
+import {ArrayComparator} from '../util/ArrayComparator';
+
+export interface IReplicatedMap<K, V> extends DistributedObject {
+    /**
+     * Associates a given value to the specified key and replicates it to the
+     * cluster. If there is an old value, it will be replaced by the specified
+     * one and returned from the call.
+     *
+     * @param key key with which the specified value is to be associated.
+     * @param value value to be associated with the specified key.
+     * @param ttl ttl to be associated with the specified key-value pair.
+     * @param unit
+     * @return old value if there was any, `null` otherwise.
+     */
+    put(key: K, value: V, ttl: Long|number|string, unit: TimeUnit): Promise<V>;
```

Comment:
😆 ok, I will add documentation note, I've seen this timeunit inside hazelcast java so thought that you mean exactly this TimeUnit class :)

`string` indeed is good option for users, because it let people who are using bigint libraries to pass the result of these libs(usually they output strings).